source/include/TypeAndDataManager.h

```diff
@@ -508,21 +508,23 @@ class TypeCommon
     UInt16  _aqAlignment:8;     // (8-15)
 
     UInt16  _encoding:kEncodingBitFieldSize;  // aggregate or single format
-    UInt16  _isFlat:1;          // (0) All data is contained in TopAQ elements ( e.g. no pointers)
-    UInt16  _isValid:1;         // (1) Contains no invalid types
-    UInt16  _isBitLevel:1;      // (2) Is a bitblock or bitcluster
+    UInt16  _isFlat:1;          // (5) All data is contained in TopAQ elements ( e.g. no pointers)
+    UInt16  _isValid:1;         // (6) Contains no invalid types
+    UInt16  _isBitLevel:1;      // (7) Is a bitblock or bitcluster
 
-    UInt16  _hasCustomDefault:1;  // (3) A non 0 non nullptr value
-    UInt16  _isMutableValue:1;    // (4) "default" value can be changed after creation.
-    UInt16  _isTemplate:1;        // (5) The type contains some generic types
-    UInt16  _hasPadding:1;        // (6) To satisfy alignment req. for elements TopAQSize() includes some padding
+    UInt16  _hasCustomDefault:1;  // (8) A non 0 non nullptr value
+    UInt16  _isMutableValue:1;    // (9) "default" value can be changed after creation.
+    UInt16  _isTemplate:1;        // (10) The type contains some generic types
+    UInt16  _hasPadding:1;        // (11) To satisfy alignment req. for elements TopAQSize() includes some padding
 
     //  properties unique to prototype elements. they are never merged up
-    UInt16  _elementUsageType:3;  // (7-9) ElementType::UsageType
+    UInt16  _elementUsageType:3;  // (12-14) ElementType::UsageType
+    UInt16  _isDataItem:1;        // (15) Element keeps track of updatedNeeded state for read/writes
+    UInt16  _needsUpdate:1;      // (0) Value has been written, needs display (tested by JS)
     //  properties unique to DefaultPointerType objects
-    UInt16  _pointerType:3;       // (10-12)
-    UInt16  _ownsDefDefData:1;    // (13) Owns DefaultDefault data (clusters and arrays)
-    UInt16  _opaqueReference:1;   // (14) Data is not an instance of the type it wraps or is templated from (e.g. refnum(Queue))
+    UInt16  _pointerType:3;       // (1-3)
```

Comment:
Great, thanks for fixing these offsets in the comment.

Reply:
It was all @spathiwa :)